crawl-ref/source/dat/des/variable/float.des

```diff
@@ -9930,3 +9957,138 @@ x..x..x11xx...........xx11x..x..x
 x1.x.1x11*xxx...{...xxx*11x1.x.1x
 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 ENDMAP
+
+NAME:   amcnicky_float_guardroom
+DEPTH:  D:3-, Depths
+ORIENT: float
+TAGS:   no_item_gen no_monster_gen
+NSUBST:  0 = 4:. / 8:0 / 6:09 / 2:98
+#Total weight 100
+NSUBST:  % = *=.:20 $:40 %:26 *:10 |:4
+NSUBST:  p = 1:+ / *=.+
+MAP
+    xxxxxxxxxx
+    x%%999%%%x
+  ..x........x
+  ..x00....00x
+xxxpx00....00x
+x00.xxxx+xxxxxxx
+x00............x
+xxxxxxpxxxxxxxpx
+           x00.x
+           x00.x
+           x00..
+           xx...
+            xx..
+ENDMAP
+
+NAME:   amcnicky_float_beach_large
+WEIGHT: 5 (D:3-), 2 (Depths)
+DEPTH:  D:3-, Depths
+ORIENT: float
+TAGS:   no_item_gen no_monster_gen
+FTILE:  . = floor_sand
+FTILE:  ; = floor_pebble
+FTILE:  0 = floor_sand
+FTILE:  9 = floor_sand
+FTILE:  % = floor_sand
+NSUBST:  0 = 4:. / 4:0 / 1:09 / 1:98
+#Total weight 100
+NSUBST:  % = *=.:20 $:40 %:26 *:10 |:4
+NSUBST:  p = 1:+ / *=.+
+SUBST:   q : .+x
+MAP
+  xxxx
+  +;;x
+  xx;x
+xxxx;xxxx
+x..0.0..x
+x.0.0.0.x
+q..www..x
+q.0www..x
+q..www..x
+x0.www..x
+x..www9.x
+x.0www..x
+x..www..x
+q0.www..x
+q..www..x
+q.0...%%x
+x.....%%x
+xxxxxxxxx
+ENDMAP
+
+NAME:   amcnicky_float_corridor_feature
+DEPTH:  D:5-, Depths, Crypt
+ORIENT: float
+TAGS:   extra no_item_gen transparent
+# Total weight: 100
+SUBST:  p: .:6 G:40 t:20 <:10 >:10 n:4 l:4 $:4 Y:2
+MAP
+xx@xx
+xp.px
+xx.xx
+xx.xx
+xp.px
+xx.xx
+xx.xx
+xp.px
+xx@xx
+ENDMAP
+
+# Vault walls and blood hint that something tougher awaits
+NAME:   amcnicky_float_rare_duel
+WEIGHT: 2
+DEPTH:  D:5-, Depths, Crypt, Lair
+ORIENT: float
+TAGS:   no_pool_fixup extra
+KPROP:  p = bloody
+NSUBST: %= 1:*:7 |:3 / 1:.:5 %:3 *:2
+NSUBST: E= 1:98 / 1:9.
+MAP
+       vvv
+      vv%v
+     vvG.vv
+ xxvvvG.E.v
+@+$pp+..8.v
+ xxvvvG.E.v
+     vvG.vv
+      vv%v
+       vvv
+ENDMAP
+
+# Involves lava, so using thematic weightings to
+# lean towards where lava seems most common in Depths
+NAME: amcnicky_float_rare_at_lavas_edge
```

Comment:
This gets away from corridor-tactics, but the terrain again doesn't provide much challenge: the player can lure the threats out of the vault easily without fighting too many at once. There's no hook to draw them into the fight.

Lava lakes are best avoided if the player can fly over them: we have them in old vaults but they encourage flight cheese or (if the player can't see the edge) flight to hunt for hidden areas/spoiler checking, so it's best not to add new ones.